lib/Data/Decimate.pm

```diff
@@ -0,0 +1,235 @@
+package Data::Decimate;
+
+use strict;
+use warnings;
+
+use 5.010;
+use Moose;
+
+use Date::Utility;
+
+use MooseX::Types::Moose qw(Int Num Str);
+use MooseX::Types -declare => [qw(
+        interval
+        )];
+
+use Moose::Util::TypeConstraints;
+use Time::Duration::Concise;
+
+subtype 'interval', as 'Time::Duration::Concise';
+coerce 'interval', from 'Str', via { Time::Duration::Concise->new(interval => $_) };
```

Comment:
I think you can drop this dependencies too in the sake of generalisation. 

1. The interval should be just a number, not number with units. I.e. it can be milliseconds, microseconds, years, or billions of years. 

2. The only constraint  put on the interval, should be: it is positive integer.